napari/_qt/_tests/test_async_slicing.py

```diff
@@ -129,3 +174,17 @@ def assert_vispy_image_data_equal() -> None:
         np.testing.assert_allclose(data, expected_data)
 
     qtbot.waitUntil(assert_vispy_image_data_equal)
+
+
+def wait_until_vispy_points_data_equal(
+    qtbot, vispy_layer: VispyImageLayer, expected_data: np.ndarray
+) -> None:
+    def assert_vispy_points_data_equal() -> None:
+        positions = vispy_layer.node._subvisuals[0]._data['a_position']
+        # Flip the coordinates because vispy uses xy instead of rc ordering.
+        # Also only take the number of dimensions expected since vispy points
+        # are always 3D even when displaying 2D slices.
+        data = positions[:, -expected_data.shape[1] :: -1]
```

Comment:
its these nuances that make coding in new areas so challenging for me (or other noobs). Thanks for the comment here!